### PR TITLE
fix: annotation broken

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY

Currently, the annotation on the time series chart has been broken.

image

1. This error is from the TypeScript [type guard function](https://github.com/apache/superset/blob/e33164024a17bb7e6de9d40b9fc60996dc9406fc/superset-frontend/packages/superset-ui-core/src/query/types/AnnotationLayer.ts#L183).
2. The annotation on the master branch doesn't apply annotation_data from the query response.

After these fixes it turns out that the Apply button is not working again, which I think should be fixed in a seprated PR. Also, the annnotation layer logic is very difficult to maintain, I would suggest investing more time in rewriting the current codes.

#### After

https://user-images.githubusercontent.com/2016594/177987046-22838736-7cde-452a-afd2-26177bf7f74d.mov

#### Before
image

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by john-bodley** [comment]
Thanks @zhaoyongjie for the fix and adding unit tests. I'm running into the exact same issue whilst testing a prior Preset release internally at Airbnb and was wondering whether you knew what PR introduced said regression?

File: superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/transformProps.ts [hunk 1]
```diff
@@ -93,12 +105,12 @@ export default function transformProps(
     queriesData,
     datasource,
     theme,
-    annotationData = {},
   } = chartProps;
   const { verboseMap = {} } = datasource;
   const [queryData] = queriesData;
   const { data = [] } = queryData as TimeseriesChartDataResponseResult;
   const dataTypes = getColtypesMapping(queryData);
+  const annotationData = getAnnotationData(chartProps);
```
Comment (stephenLYZ) [inline superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/transformProps.ts:113]: Mixd-timeseries chart may also need to revert here.

Comment (zhaoyongjie) [reply superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/transformProps.ts:104]: done! thanks for the heads up.

File: superset-frontend/plugins/plugin-chart-echarts/src/utils/annotation.ts [hunk 2]
```diff
@@ -130,3 +134,13 @@ export function extractAnnotationLabels(
...
 }
+
+export function getAnnotationData(
+  chartProps: EchartsTimeseriesChartProps | EchartsMixedTimeseriesProps,
+): AnnotationData {
+  const data = chartProps?.queriesData[0]?.annotation_data as AnnotationData;
+  if (!isEmpty(data)) {
+    return data;
+  }
+  return {};
```
Comment (ktmud) [inline superset-frontend/plugins/plugin-chart-echarts/src/utils/annotation.ts:145]: Nit: ternary seems more readable here

```suggestion
    return isEmpty(data) ? {} : data;
```

I'm also not sure `isEmpty` is needed. If for whatever reason `annotation_data` becomes anything other than a plain object, it probably means there's bugs in the code or corrupted data and we'd need to fix it.

**Review by ktmud — APPROVED** [review]
LGTM

**Comment by zhaoyongjie** [comment]
@rusackas the coverage issue didn't bring from this PR, the master branch has broken some days. I will look into / fix coverage issue in the separated pr.

File: superset-frontend/plugins/plugin-chart-echarts/src/utils/annotation.ts (hunk 2 above)
Comment (zhaoyongjie) [reply superset-frontend/plugins/plugin-chart-echarts/src/utils/annotation.ts:145]: @ktmud Thanks, i will look into this case.

**Comment by slavanorm** [comment]
can we say that this issue is not complete?
in 2.* version annotations are broken:
https://github.com/apache/superset/issues/24374

**Pull request merged**